file-share-service-desktop-client/FileShareApiAdminClientFactory.cs

```diff
@@ -40,24 +45,41 @@ public IFileShareApiAdminClient Build()
     public class UserAgentClientFactory : IHttpClientFactory
     {
         private readonly IVersionProvider versionProvider;
+        private readonly ILogger<IFileShareApiAdminClientFactory> logger;
 
-        public UserAgentClientFactory(IVersionProvider versionProvider)
+        public UserAgentClientFactory(IVersionProvider versionProvider, ILogger<IFileShareApiAdminClientFactory> logger)
         {
             this.versionProvider = versionProvider;
+            this.logger = logger;
         }
 
         public HttpClient CreateClient(string name)
         {
-            var client = new HttpClient();
-            client.DefaultRequestHeaders.UserAgent.Add(new ProductInfoHeaderValue("FileShareServiceDesktopClient",
-                versionProvider.Version));
-            client.DefaultRequestHeaders.UserAgent.Add(new ProductInfoHeaderValue(nameof(FileShareApiAdminClient),
-                typeof(FileShareApiAdminClient)
+            int retryCount = Convert.ToInt32(System.Configuration.ConfigurationManager.AppSettings["RetryCount"]);
```

Comment:
Instead of creating a new service provider here, please use the existing service provider that's already configured in the `App` class `CreateContainerExtension()` method.

The code in `CreateContainerExtension()` should be something like this:
````
            var serviceCollection = new ServiceCollection();

            serviceCollection.AddLogging(loggingBuilder => {
                var logger = new LoggerConfiguration()
                        .ReadFrom.AppSettings()
                        .CreateLogger();
                loggingBuilder.AddSerilog(logger, dispose: true);
            });

            int retryCount = Convert.ToInt32(ConfigurationManager.AppSettings["RetryCount"]);
            int sleepDurationMultiplier = Convert.ToInt32(ConfigurationManager.AppSettings["SleepDurationMultiplier"]);
            const string FSSClient = "FSSClient";

            serviceCollection.AddHttpClient(FSSClient)
                .AddPolicyHandler((services, request) => {
                    var logger = services.GetService<ILogger<IFileShareApiAdminClientFactory>>();
                    return TransientErrorsHelper.GetRetryPolicy(logger, "FileShareApiAdminClient", retryCount, sleepDurationMultiplier, out _);
                 });
````

The `IServiceProvider` can then be injected into the ctor for the `FileShareApiAdminClientFactory`
````
public FileShareApiAdminClientFactory(IEnvironmentsManager environmentsManager, IAuthTokenProvider authTokenProvider,
            IVersionProvider versionProvider, ILogger<IFileShareApiAdminClientFactory> logger, IServiceProvider sp)
````

Reply:
And then the code in `CreateClient()` can simplify to:
````
public HttpClient CreateClient(string name)
        {
            var configuredClient = sp.GetRequiredService<IHttpClientFactory>()
                    .CreateClient("FSSClient");

            configuredClient.DefaultRequestHeaders.UserAgent.Add(
            new ProductInfoHeaderValue("FileShareServiceDesktopClient", versionProvider.Version));
            configuredClient.DefaultRequestHeaders.UserAgent.Add(new ProductInfoHeaderValue(nameof(FileShareApiAdminClient), typeof(FileShareApiAdminClient)
                    .Assembly
                    .GetCustomAttributes<AssemblyFileVersionAttribute>()
                    .Single()
                    .Version));

            return configuredClient;
        }
````